lib/client.js

```diff
@@ -582,18 +582,19 @@
     }
   });
 
+  HappnClient.prototype.__writeCookie = function(session, sessionDocument) {
+    var cookieName = session.cookieName || 'happn_token';
+    if (this.options.protocol === 'https') cookieName += '_https';
```

Comment:
This will have backwards compatibility issues with frontend code expecting a certain cookie name - on SMC Cloud the cookie name is 'FieldPoPDeviceCloud' and it is used in its frontend code in some places. I think this must be a option we enable - ideally server side.

Reply:
Can session creation in this line not decide whether to add the _https or not when it responds:
https://github.com/happner/happn-3/blob/ce1fee5a084d0b6782c983494f61acb3751913f6/lib/services/security/service.js#L1302

Then we always just work with the cookieName as detected when the session is created.

We have to add a new config parameter, something like cookieNameAppendProtocol, to decide whether to append the https or not. This has to default to false.

The session creation must then check the protocol AND whether cookieNameAppendProtocol is set.

Reply:
this has been done, kind of, instead of pushing the final cookieName to the client, the httpsCookie configuration setting is pushed to the client, which then decides to append https or not, based on whether the client is connecting via HTTP or https, httpsCookie is false by default

Reply:
This has been done, fully